packages/notifi-react-card/lib/components/NotifiSmsInput.tsx

```diff
@@ -76,52 +84,59 @@ export const NotifiSmsInput: React.FC<NotifiSmsInputProps> = ({
   }, [phoneNumber, splitPhoneValues]);
 
   useEffect(() => {
-    if (phoneValues.baseNumber === '') {
+    if (phoneValues.baseNumber === '' && phoneNumber !== '') {
+      console.log('phone number', phoneNumber);
       setPhoneNumber('');
     }
-  }, [phoneValues.baseNumber, setPhoneNumber]);
+  }, [phoneValues.baseNumber, setPhoneNumber, phoneNumber]);
 
-  const countryCodes = allowedCountryCodes.map((code) => {
-    return (
-      <option
-        className={clsx(
-          'NotifiSmsInput__dropdownOption',
-          classNames?.dropdownOption,
-        )}
-        key={code}
-        value={code}
-      >
-        {code}
-      </option>
-    );
-  });
+  const countryCodes = useMemo(() => {
+    return allowedCountryCodes.map((code) => {
+      return (
+        <option
+          className={clsx(
+            'NotifiSmsInput__dropdownOption',
+            classNames?.dropdownOption,
+          )}
+          key={code}
+          value={code}
+        >
+          {code}
+        </option>
+      );
+    });
+  }, [allowedCountryCodes, classNames]);
 
-  const validateSmsInput = () => {
+  const validateSmsInput = useCallback(() => {
     if (phoneNumber === '') {
       return;
     }
 
     if (!isValidPhoneNumber(phoneNumber)) {
-      setSmsErrorMessage('The phone number is invalid. Please try again');
+      setPhoneNumberErrorMessage(
+        'The phone number is invalid. Please try again',
+      );
     }
-  };
+  }, [phoneNumber, setPhoneNumberErrorMessage]);
 
-  const handleChange =
+  const handleChange = useCallback(
     (field: 'dialCode' | 'baseNumber') =>
-    (event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
-      if (field === 'baseNumber') {
-        const onlyNumberInput = event.target.value.replace(/[^\d]/g, '');
+      (event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+        if (field === 'baseNumber') {
+          const onlyNumberInput = event.target.value.replace(/[^\d]/g, '');
 
-        setPhoneValues({ ...phoneValues, [field]: onlyNumberInput });
-      } else {
-        setPhoneValues({ ...phoneValues, [field]: event.target.value });
-      }
-      if (field === 'dialCode') {
-        setPhoneNumber(event.target.value + phoneValues.baseNumber);
-      } else {
-        setPhoneNumber(phoneValues.dialCode + event.target.value);
-      }
-    };
+          setPhoneValues({ ...phoneValues, [field]: onlyNumberInput });
+        } else {
+          setPhoneValues({ ...phoneValues, [field]: event.target.value });
+        }
+        if (field === 'dialCode') {
```

Comment:
the fact that we have `if (field)` here makes me think that we should just have two different functions, handleDialCodeChange and handleBaseNumberChange, rather than parameterizing the field (which results in a different function instance actually used for the event handler on every render)

Reply:
genius